server/src/main/java/io/spine/system/server/MirrorProjection.java

```diff
@@ -0,0 +1,220 @@
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.system.server;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Message;
+import io.spine.client.CompositeColumnFilter;
+import io.spine.client.EntityFilters;
+import io.spine.client.EntityId;
+import io.spine.client.EntityIdFilter;
+import io.spine.client.Target;
+import io.spine.core.Subscribe;
+import io.spine.server.entity.LifecycleFlags;
+import io.spine.server.entity.storage.Column;
+import io.spine.server.projection.Projection;
+import io.spine.type.TypeUrl;
+
+import java.util.Collection;
+import java.util.List;
+
+import static io.spine.client.ColumnFilters.all;
+import static io.spine.client.ColumnFilters.eq;
+import static io.spine.protobuf.AnyPacker.pack;
+import static io.spine.protobuf.AnyPacker.unpack;
+import static io.spine.server.entity.FieldMasks.applyMask;
+import static io.spine.validate.Validate.isDefault;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * The {@link Mirror} projection for domain aggregates.
+ *
+ * <p>This entity belongs to the system context. It mirrors the state of an aggregate. Not every
+ * aggregate has a mirror projection. Refer to the {@link MirrorRepository} for more details.
+ *
+ * <p>The projection defines an {@link io.spine.server.entity.storage.EntityColumn EntityColumn}
+ * which stored the {@linkplain #getAggregateType() type URL} of the state, which is mirrored.
+ *
+ * @author Dmytro Dashenkov
+ */
+public final class MirrorProjection extends Projection<MirrorId, Mirror, MirrorVBuilder> {
+
+    private static final String TYPE_COLUMN_NAME = "aggregate_type";
+    private static final String TYPE_COLUMN_QUERY_NAME = "aggregateType";
+
+    MirrorProjection(MirrorId id) {
+        super(id);
+    }
+
+    @Subscribe
+    public void on(EntityStateChanged event) {
+        getBuilder().setId(getId())
+                    .setState(event.getNewState());
+    }
+
+    @Subscribe
+    public void on(@SuppressWarnings("unused")
```

Comment:
Please have a comment on suppressions. Perhaps, in the `@implNote` section on a class-level.